config/database.yml

```diff
@@ -1,20 +1,35 @@
-<% user = ENV.key?("POSTGRESQL_ADMIN_PASSWORD") ? "postgres" : ENV["POSTGRESQL_USER"] %>
-<% password = ENV.key?("POSTGRESQL_ADMIN_PASSWORD") ? ENV["POSTGRESQL_ADMIN_PASSWORD"] : ENV["POSTGRESQL_PASSWORD"] %>
-<% db_service = ENV.fetch("DATABASE_SERVICE_NAME","").upcase.sub("-", "_") %>
+<%
+  if ClowderCommonRuby::Config.clowder_enabled?
 
+    config = ClowderCommonRuby::Config.load
+
+    username = config.database.username
+    password = config.database.password
+    host = config.database.hostname
+    database = config.database.name
+  else
+
+    db_service = ENV.fetch("DATABASE_SERVICE_NAME","").upcase.sub("-", "_")
+
+    username = ENV.key?("POSTGRESQL_ADMIN_PASSWORD") ? "postgres" : ENV["POSTGRESQL_USER"]
+    password = ENV.key?("POSTGRESQL_ADMIN_PASSWORD") ? ENV["POSTGRESQL_ADMIN_PASSWORD"] : ENV["POSTGRESQL_PASSWORD"]
+    host = ENV.fetch("#{db_service}_SERVICE_HOST")
+    database = Rails.env.test? ? ENV['POSTGRESQL_TEST_DATABASE'] || ENV["POSTGRESQL_DATABASE"] : ENV["POSTGRESQL_DATABASE"]
+  end
+ %>
```

Comment:
This... shouldn't work? These are local variables defined within the if/else scope and should be not defined below when you're using them. You could keep the existing three lines at the top and just override the variables if clowder is enabled.

Reply:
These variables are all within the scope of the YAML/whole template. Ruby doesn't scope within if or begin blocks, as for example JS does with `let`.

Reply:
Hmm! All this time, I didn't realize that :laughing: